calc/CalcEvaluator.scala

```diff
@@ -0,0 +1,98 @@
+package distcompiler.calc
+
+import cats.syntax.all.given
+
+import distcompiler.*
+import dsl.*
+import distcompiler.Builtin.{Error, SourceMarker}
+import distcompiler.calc.tokens.*
+
+object CalcEvaluator extends PassSeq:
+  import distcompiler.dsl.*
+  import Reader.*
+  import CalcReader.*
+
+  def inputWellformed: Wellformed = CalcParser.outputWellformed
+
+  private val simplifyPass = passDef:
+    wellformed := inputWellformed.makeDerived:
+      Node.Top ::=! fields(Number)
+
+    pass(once = false, strategy = pass.topDown)
+      .rules:
+        on(
+          field(tok(Expression)) *> children(
+            tok(Add).product(children(
+              field(tok(Number))
```

Comment:
how to find Expression with nested Number specifically:

```scala
field(tok(Expression) *> onlyChild(tok(Number)))
```

more general alternative
```scala
field(tok(Expression).withChildren(field(tok(Number)) ~ eof))
```